library/index.c

```diff
@@ -323,24 +323,25 @@ int eblob_index_blocks_fill(struct eblob_base_ctl *bctl)
 }
 
 
-static struct eblob_disk_control *eblob_find_on_disk(struct eblob_backend *b,
-		struct eblob_base_ctl *bctl, struct eblob_disk_control *dc,
+static int eblob_find_on_disk(struct eblob_backend *b,
```

Comment:
end = bctl->sort.data + bctl->sort.size;
start = bctl->sort.data;
...
while (sorted < end && eblob_disk_control_sort(sorted, dc) == 0) {

So it may do linear search through whole index file, not only in single index block. Index file may contain keys with same id.